### ISSUE TITLE
Calculating cross-correlation after dropping nan's

### ISSUE THREAD

**Issue by ramaroesilva** [LDSSA/batch4-students#49]
In the Learning Notebook 2, the following procedure is suggested for calculating the pacf of a timeseries with nan values (and the same could apply for cross-correlations): plot_acf(df.dropna()) 

Or in more explicit terms: i) drop the nan's; ii) calculate the cross-correlation.

However, when we drop the nan's, the temporal meaning of a lag becomes inconsistent throughout the timeseries (i.e. whenever there's a "data hole" the actual lag between two data points will be >1).

**Comment by ramaroesilva** [comment]
My suggestion... loop through the various lag terms you want to be considered and inside the loop:
i) create a df with the original timeseries and the given lagged term
ii) delete the rows that have nan values in any of the 2 columns
iii) calculate correlation between the two columns

that way you effectively drop the nan's, but keep the correct original-lag term correspondence

**Comment by ramaroesilva** [comment]
Suggestion from Telmo Felgueira:

«Something fun to try is grab a full dataset and plotting the ACF. Then remove one sample and plot the ACF in two ways:
Doing ffill
Dropping the nan

It would be fun to see which one look more like the original.»

**Comment by Juliana-R** [comment]
@ramaroesilva thanks for the feedback! Can you specify the Learning Unit this refers to?

**Comment by ramaroesilva** [comment]
@Juliana-R, yes I should have been more clear. just edited my first comment accordingly :-)
